spec/models/reply_spec.rb

```diff
@@ -173,4 +173,35 @@
       expect(post.replies.ordered).to eq([first_reply, third_reply, second_reply])
     end
   end
+
+  describe "#destroy_subsequent_replies" do
+    it "works" do
+      post = create(:post)
+      replies = create_list(:reply, 2, post: post)
+      reply = create(:reply, post: post)
+      create_list(:reply, 2, post: post)
+      expect { reply.send(:destroy_subsequent_replies) }.to change { Reply.count }.by(-3)
+      expect(Reply.where(id: replies.map(&:id)).count).to eq(2)
+      expect(Reply.find_by(id: reply.id)).not_to be_present
+      expect(post.reload.last_reply_id).to eq(reply.id)
```

Comment:
This seems strange, but maybe I'm misreading something – the reply with `id: reply.id` doesn't exist (per line 185), but here, the `last_reply_id` of the post is this reply? Isn't that a broken foreign key?

Reply:
Er. yeah it should not in fact do that, which makes me wonder why tests are passing!